aggregator/src/binary_utils.rs

```diff
@@ -347,6 +353,70 @@ async fn health_endpoint_server(address: SocketAddr) {
         .await;
 }
 
+fn zpages_handler(trace_reload_handle: TraceReloadHandle) -> impl Handler {
+    (
+        Head::new(),
+        State(Arc::new(trace_reload_handle)),
+        Router::new()
+            .get(
+                "/healthz",
+                |conn: trillium::Conn| async move { conn.ok("") },
+            )
+            .get("/traceconfigz", api(get_traceconfigz))
+            .put("/traceconfigz", api(put_traceconfigz)),
+    )
+}
+
+async fn get_traceconfigz(
+    conn: &mut trillium::Conn,
+    State(trace_filter_handler): State<Arc<TraceReloadHandle>>,
+) -> Result<Json<TraceconfigzBody>, Status> {
+    Ok(Json(TraceconfigzBody {
+        filter: trace_filter_handler
+            .with_current(|trace_filter| trace_filter.to_string())
+            .map_err(|err| {
+                conn.set_body(format!("failed to get current filter: {}", err));
+                Status::InternalServerError
+            })?,
+    }))
+}
+
+async fn put_traceconfigz(
+    conn: &mut trillium::Conn,
+    (State(trace_filter_handler), Json(req)): (
+        State<Arc<TraceReloadHandle>>,
+        Json<TraceconfigzBody>,
+    ),
+) -> Result<Json<TraceconfigzBody>, Status> {
+    trace_filter_handler
+        .reload(EnvFilter::new(req.filter))
+        .map_err(|err| {
+            conn.set_body(format!("failed to update filter: {}", err));
+            // Note: reload will accept malformed filter directives, and fall back to the `error`
+            // filter. An error indicating malformation is never propagated up, so we can't give
+            // a proper HTTP status code. The error state will be revealed to the operator in the
+            // response body, however, since it will show the fallback log filter rather than their
+            // desired one.
```

Comment:
We can improve this by using `EnvFilter::try_new()`, which only falls back to `error` on empty inputs, not invalid inputs. The errors returned by `reload()` itself indicate that an internal lock was poisoned by a panic or the reloadable subscriber was dropped, so a 500 status code sounds good for them.